inmem.go

```diff
@@ -0,0 +1,244 @@
+package clover
+
+import (
+	"sort"
+	"sync"
+)
+
+type collection map[string]*Document
+
+type inMemEngine struct {
+	sync.RWMutex
+	collections map[string]collection
+}
+
+func newInMemoryStoreEngine() StorageEngine {
+	return &inMemEngine{
+		RWMutex:     sync.RWMutex{},
+		collections: make(map[string]collection),
+	}
+}
+
+func InMemoryDB() (*DB, error) {
+	return &DB{
+		dir:    "",
+		engine: newInMemoryStoreEngine(),
+	}, nil
+}
+
+// Close implements StorageEngine
+func (*inMemEngine) Close() error {
+	return nil
+}
+
+// CreateCollection implements StorageEngine
+func (e *inMemEngine) CreateCollection(name string) error {
+	e.Lock()
+	defer e.Unlock()
+	if ok, _ := e.HasCollection(name); ok {
+		return ErrCollectionExist
+	}
+	e.collections[name] = make(collection)
+	return nil
+}
+
+// Delete implements StorageEngine
+func (e *inMemEngine) Delete(q *Query) error {
```

Comment:
I wanted you to use function `replaceDocs` because this can be used to implement both `Update` and `Delete`.
So `Delete()` can be simply implemented in the following way:

```go
func (e *inMemEngine) Delete(q *Query) error {
	return e.replaceDocs(nil, q, func(_ *Document) *Document {
		return nil
	})
}
```